recipes/vtk/all/conanfile.py

```diff
@@ -0,0 +1,197 @@
+            tools.patch(**patch)
+
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        # "$\package\lib\vtk" contains "hierarchy\conanvtk\" and a lot of *.txt files in it. I believe they are not needed. Remove them.
+        tools.rmdir(os.path.join(self.package_folder, 'lib', 'vtk'))
+        # "$\package\lib\cmake" contains a lot of *.cmake files. conan-center HOOK disallow *.cmake files in package. Remove them.
+        tools.rmdir(os.path.join(self.package_folder, 'lib', 'cmake'))
+        # Licenses are created in "$\package\share\licenses\conanvtk\" while their must be in "$\package\licenses\". Move them from former to latter.
+        os.rename(os.path.join(self.package_folder, 'share', 'licenses', 'conanvtk'), os.path.join(self.package_folder, 'licenses'))
+        tools.rmdir(os.path.join(self.package_folder, 'share'))
+
+    # GCC static linking require providing libraries in appropriate order.
+    # Below "order" is not necessary is correct (couldn't find the correct one), but it is used as good starting point.
+    def _sort_libs(self, libs):
+        if self.settings.compiler != "gcc":
+            return libs
+        libs_ordered = []
+        print('VTK libs before sort: ' + (';'.join(libs)))
+        order = ['vtkViewsCore', 'vtkViewsContext2D', 'vtkRenderingVolumeOpenGL2', 'vtkRenderingContextOpenGL2', 'vtkRenderingContext2D', 'vtkRenderingVolume', 'vtkfreetype', 'vtkRenderingFreeType', 'vtkRenderingAnnotation', 'vtkInteractionWidgets', 'vtkInteractionStyle', 'vtkImagingMath', 'vtkImagingHybrid', 'vtkImagingGeneral', 'vtkImagingColor', 'vtkexpat', 'vtkIOXMLParser', 'vtkIOXML', 'vtkIOLegacy', 'vtkIOGeometry', 'vtkgl2ps', 'vtkglew', 'vtkRenderingOpenGL2', 'vtkRenderingGL2PSOpenGL2', 'vtktiff', 'vtkpng', 'vtkjpeg', 'vtkmetaio', 'vtkIOImage', 'vtkIOExport', 'vtkzlib', 'vtkIOCore', 'vtkFiltersModeling', 'vtkFiltersSources', 'vtkRenderingCore', 'vtkImagingSources', 'vtkFiltersHybrid', 'vtkFiltersGeometry', 'vtkalglib', 'vtkImagingCore', 'vtkImagingFourier', 'vtkFiltersStatistics', 'vtkFiltersGeneral', 'vtkFiltersExtraction', 'vtkFiltersCore', 'vtkDICOMParser', 'vtkCommonExecutionModel', 'vtkCommonComputationalGeometry', 'vtkCommonDataModel', 'vtkCommonSystem', 'vtkCommonTransforms', 'vtkCommonMath', 'vtkCommonMisc', 'vtkCommonCore', 'vtkCommonColor', 'vtksys']
+        for item in order:
+            for idx in range(len(libs)): 
+                if item.lower() in libs[idx].lower():
+                    value = libs.pop(idx)
+                    libs_ordered.append(value)
+                    break
+        libs_ordered = libs + libs_ordered # add unordered elements, if any
+        print('VTK libs ordered: ' + (';'.join(libs_ordered)))
+        return libs_ordered
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "VTK"
```

Comment:
Can you check what cmake modules vtk installs and recreate them using components?

Reply:
@madebr 
Not sure what are You taking about.
Do You mean more or less what was done in [Added VTK modules and groups](https://github.com/conan-io/conan-center-index/pull/3280/commits/a0dae91739e699995e8e138c9f806f87f5637e71) commit?

Reply:
In the `package` method, at the line 
```python
tools.rmdir(os.path.join(self.package_folder, 'lib', 'cmake'))
```
you remove the cmake config scripts.
The hooks indeed do block  `XXX-config.cmake` and `FindXXX.cmake` scripts.
But when a consumers needs VTK in his cmake script, he still want to do something like the following:
```cmake
find_package(VTK CONFIG REQUIRED COMPONENTS Opt1 OPT2)
target_link_library(my_exe PRIVATE VTK::Opt1 VTK::Opt2)
```
Using conan components allows to define these cmake imported targets (to a certain extent).